modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -4026,95 +4026,162 @@ public void testGetExtensionDynamicObjectDefinitionTableValues()
-			new LongIntegerObjectFieldBuilder(
-			).labelMap(
-				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
-			).name(
-				"longField"
-			).objectDefinitionId(
-				objectDefinition.getObjectDefinitionId()
-			).build());
-		ObjectField objectField2 = _addCustomObjectField(
-			new TextObjectFieldBuilder(
-			).labelMap(
-				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
-			).name(
-				"textField"
-			).objectDefinitionId(
-				objectDefinition.getObjectDefinitionId()
-			).required(
-				true
-			).build());
-
-		User user = UserTestUtil.addUser();
+		boolean originalEnableLocalization =
+			objectDefinition.isEnableLocalization();
 
 		try {
+			objectDefinition.setEnableLocalization(true);
+
+			objectDefinition =
+				_objectDefinitionLocalService.updateObjectDefinition(
+					objectDefinition);
+
+			ObjectField objectField1 = _addCustomObjectField(
+				new TextObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).localized(
+					true
+				).name(
+					"localizedTextField"
+				).objectDefinitionId(
+					objectDefinition.getObjectDefinitionId()
+				).build());
+			ObjectField objectField2 = _addCustomObjectField(
+				new LongIntegerObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).name(
+					"longField"
+				).objectDefinitionId(
+					objectDefinition.getObjectDefinitionId()
+				).build());
+			ObjectField objectField3 = _addCustomObjectField(
+				new TextObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).name(
+					"textField"
+				).objectDefinitionId(
+					objectDefinition.getObjectDefinitionId()
+				).required(
+					true
+				).build());
+
+			User user = UserTestUtil.addUser();
+
+			try {
+				_objectEntryLocalService.
+					addOrUpdateExtensionDynamicObjectDefinitionTableValues(
+						TestPropsValues.getUserId(), objectDefinition,
+						user.getUserId(), Collections.emptyMap(),
+						ServiceContextTestUtil.getServiceContext());
+
+				Assert.fail();
+			}
+			catch (ObjectEntryValuesException.Required
+						objectEntryValuesException) {
+
+				Assert.assertEquals(
+					"No value was provided for required object field " +
+						"\"textField\"",
+					objectEntryValuesException.getMessage());
+			}
+
+			Map<String, String> localizedValues = HashMapBuilder.put(
+				"en_US", RandomTestUtil.randomString()
+			).put(
+				"pt_BR", RandomTestUtil.randomString()
+			).build();
+
+			Map<String, Serializable> values =
+				HashMapBuilder.<String, Serializable>put(
+					objectField1.getI18nObjectFieldName(),
+					(Serializable)localizedValues
+				).put(
+					objectField1.getName(),
+					localizedValues.get(
+						LocaleUtil.toLanguageId(LocaleUtil.getDefault()))
+				).put(
+					"longField", 10L
+				).put(
+					"textField", "Value"
+				).build();
```

Comment:
@pedro-oliveira446 inline:

```java
Map<String, Serializable> values =
	HashMapBuilder.<String, Serializable>put(
		"localizedTextField", "en_US localizedTextFieldValue1"
	).put(
		"localizedTextField_i18n",
			HashMapBuilder.put(
				"en_US", "en_US localizedTextFieldValue1"
			).put(
				"pt_BR", "pt_BR localizedTextFieldValue1"
			).build()
	).put(
		"longField", 10L
	).put(
		"textField", "Value"
	).build();
```

 `longField` and `textField` are not using random values, so let's do the same for the localized field.

Reply:
Just done!